pkg/apply/applydrivers/apply_drivers_default.go

```diff
@@ -43,12 +43,21 @@ func NewDefaultApplier(cluster *v2.Cluster, cf clusterfile.Interface, images []s
 	if !cluster.CreationTimestamp.IsZero() && err != nil {
 		return nil, err
 	}
+	var RunNewImages []string
+	var ClusterUpgradeV [2]string
```

Comment:
Seems you don't need this string arr. just compare the current.Spec.Image and the desired.Spec.Image, and judge upgrade the cluster or not.